### PR TITLE
Add IMXRT1020-EVK Configuration for NXP

### PR DESCRIPTION
**Pull request by HarrisonChoi750405**
Add IMXRT1020-EVK Configuration for NXP

1. Add build configuration for iMXRT1020-EVK
2. Add Register map and IOMUXC configuration for iMXRT1020-EVK
3. Add definition for iMXRT1020-EVK to common source code.
4. Add booting sequence and flash function for iMXRT1020-EVK
5. Add download and configuration file for iMXRT1020-EVK

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [d4bd7131101c3fc8dbf110a47cd5c3c01d542f60] - Code Rule Check (C++) OK.

File: build/configs/imxrt1020-evk/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# IMXRT1050-EVK
```
[the remark this answers is not in the document]
Comment (sunghan-chang) [reply build/configs/imxrt1020-evk/README.md]: @HarrisonChoi750405 Could you fix this? Then, I will merge this

[the remark this answers is not in the document]
Comment (HarrisonChoi750405) [reply build/configs/imxrt1020-evk/README.md]: I have just fixed it

**Comment by seinfra** [comment]
Target : [0ea20aa5e6f29285f9bf83ec52f4bcb583b61009] - Code Rule Check (C++) OK.

**Pull request merged**